extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,141 @@
+import os.path
+import sys
+import typing
+
+from conan.api.output import cli_out_write, ConanOutput
+from conan.api.conan_api import ConanAPI
+from conan.cli.args import common_graph_args, validate_common_graph_args
+from conan.cli.command import conan_command
+
+if typing.TYPE_CHECKING:
+    from conans.client.graph.graph import Node
+
+
+def package_type_to_component_type(pt: str):
+    from cyclonedx.model.component import ComponentType
+    if pt is "application":
+        return ComponentType.APPLICATION
+    else:
+        return ComponentType.LIBRARY
+
+
+def licenses(ids):
```

Comment:
maybe add typing, as later it is checked for `isinstance(ids, tuple)`
```py
from typing import Iterable, Optional, Tuple, Union
def licenses(ids: Union[Tuple[str], str]) -> Optional[Iterable['LicenseChoice']]:
  ...
```

Reply:
Type annotation added, but I cannot use `'LicenseChoice'` as this is not imported at that point. The import happens inside the body of the method.

Reply:
i will pullrequested some restructuring, that you can simply merge into your PR:
then, all should work fine and allow a much easier maintenance.

PR coming soon.